src/detectors/FHCAL/FHCAL.cc

```diff
@@ -19,196 +17,6 @@ extern "C" {
-          "HcalEndcapPMergedHits", {"HcalEndcapPRecHits"}, {"HcalEndcapPMergedHits"},
-          {
-            .readout = "HcalEndcapPHits",
-            .fields = {"layer", "slice"},
-            .refs = {1, 0},
-          },
-          app   // TODO: Remove me once fixed
-        ));
-        app->Add(new JChainMultifactoryGeneratorT<CalorimeterTruthClustering_factoryT>(
-          "HcalEndcapPTruthProtoClusters", {"HcalEndcapPRecHits", "HcalEndcapPHits"}, {"HcalEndcapPTruthProtoClusters"},
-          app   // TODO: Remove me once fixed
-        ));
-        app->Add(new JChainMultifactoryGeneratorT<CalorimeterIslandCluster_factoryT>(
-          "HcalEndcapPIslandProtoClusters", {"HcalEndcapPRecHits"}, {"HcalEndcapPIslandProtoClusters"},
-          {
-            .sectorDist = 5.0 * dd4hep::cm,
-            .localDistXY = {15.0*dd4hep::mm, 15.0*dd4hep::mm},
-            .dimScaledLocalDistXY = {15.0*dd4hep::mm, 15.0*dd4hep::mm},
-            .splitCluster = true,
-            .minClusterHitEdep = 0.0 * dd4hep::MeV,
-            .minClusterCenterEdep = 30.0 * dd4hep::MeV,
-            .transverseEnergyProfileMetric = "globalDistEtaPhi",
-            .transverseEnergyProfileScale = 1.,
-          },
-          app   // TODO: Remove me once fixed
-        ));
-
-        app->Add(
-          new JChainMultifactoryGeneratorT<CalorimeterClusterRecoCoG_factoryT>(
-             "HcalEndcapPTruthClusters",
-            {"HcalEndcapPTruthProtoClusters",        // edm4eic::ProtoClusterCollection
-             "HcalEndcapPHits"},                     // edm4hep::SimCalorimeterHitCollection
-            {"HcalEndcapPTruthClusters",             // edm4eic::Cluster
-             "HcalEndcapPTruthClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
-            {
-              .energyWeight = "log",
-              .moduleDimZName = "",
-              .sampFrac = 1.0,
-              .logWeightBase = 3.6,
-              .depthCorrection = 0.0,
-              .enableEtaBounds = false
-            },
-            app   // TODO: Remove me once fixed
-          )
-        );
-
-        app->Add(
-          new JChainMultifactoryGeneratorT<CalorimeterClusterRecoCoG_factoryT>(
-             "HcalEndcapPClusters",
-            {"HcalEndcapPIslandProtoClusters",  // edm4eic::ProtoClusterCollection
-             "HcalEndcapPHits"},                // edm4hep::SimCalorimeterHitCollection
-            {"HcalEndcapPClusters",             // edm4eic::Cluster
-             "HcalEndcapPClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
-            {
-              .energyWeight = "log",
-              .moduleDimZName = "",
-              .sampFrac = 0.033,
-              .logWeightBase = 6.2,
-              .depthCorrection = 0.0,
-              .enableEtaBounds = false,
-            },
-            app   // TODO: Remove me once fixed
-          )
-        );
-
-        app->Add(new JChainMultifactoryGeneratorT<CalorimeterHitDigi_factoryT>(
```

Comment:
Don't we need to keep the insert still?

Reply:
Good catch. We should keep it, actually.